tests/quantization/torchao/test_torchao.py

```diff
@@ -0,0 +1,358 @@
+        }
+
+        return inputs
+
+    def _test_quant_type(self, quantization_config: TorchAoConfig, expected_slice: List[float]):
+        components = self.get_dummy_components(quantization_config)
+        pipe = FluxPipeline(**components)
+        pipe.to(device=torch_device, dtype=torch.bfloat16)
+
+        inputs = self.get_dummy_inputs(torch_device)
+        output = pipe(**inputs)[0]
+        output_slice = output[-1, -1, -3:, -3:].flatten()
+
+        self.assertTrue(np.allclose(output_slice, expected_slice, atol=1e-3, rtol=1e-3))
+
+    def test_quantization(self):
+        # TODO(aryan): update these values from our CI
+        QUANTIZATION_TYPES_TO_TEST = [
+            ("int4wo", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+            ("int4dq", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+            ("int8wo", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+            ("int8dq", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+            ("uint4wo", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+            ("int_a8w8", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+            ("uint_a16w7", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+        ]
+
+        if TorchAoConfig._is_cuda_capability_atleast_8_9():
+            QUANTIZATION_TYPES_TO_TEST.extend(
+                [
+                    ("float8wo_e5m2", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+                    ("float8wo_e4m3", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+                    ("float8dq_e4m3", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+                    ("float8dq_e4m3_tensor", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+                    ("float8dq_e4m3_row", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+                    ("fp4wo", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+                    ("fp6", np.array([0, 0, 0, 0, 0, 0, 0, 0, 0])),
+                ]
+            )
+
+        for quantization_name, expected_slice in QUANTIZATION_TYPES_TO_TEST:
+            quantization_config = TorchAoConfig(quant_type=quantization_name)
+            self._test_quant_type(quantization_config, expected_slice)
+
+    def test_int4wo_quant_bfloat16_conversion(self):
+        """
+        Tests whether the dtype of model will be modified to bfloat16 for int4 weight-only quantization.
+        """
+        quantization_config = TorchAoConfig("int4_weight_only", group_size=64)
+        quantized_model = FluxTransformer2DModel.from_pretrained(
+            "hf-internal-testing/tiny-flux-pipe",
+            subfolder="transformer",
+            quantization_config=quantization_config,
+            torch_dtype=torch.bfloat16,
+        )
+
+        qlayer = quantized_model.transformer_blocks[0].attn.to_q
+        weight = qlayer.weight
+        self.assertTrue(isinstance(weight, AffineQuantizedTensor))
+        self.assertEqual(weight.quant_min, 0)
+        self.assertEqual(weight.quant_max, 15)
+        self.assertTrue(isinstance(weight.layout_type, TensorCoreTiledLayoutType))
```

Comment:
@jerryzh168 I am unable to pass these tests (adapted from transformers integration). The check of weight being an `AffineQuantizedTensor` fails, but I can see from printing that the weights are indeed modified.

Reply:
can you paste what you see from printing qlayer.weight?

Reply:
This is the result for:

```
qlayer = quantized_model.transformer_blocks[0].attn.to_q
weight = qlayer.weight
print(qlayer)
print(weight)
```

```
Linear(in_features=32, out_features=32, weight=not recognized)
Parameter containing:
tensor([[-0.0811,  0.0933,  0.0227,  ..., -0.0601, -0.1455,  0.0449],
        [-0.0417, -0.0248, -0.1348,  ...,  0.0894, -0.1602,  0.0625],
        [-0.1279,  0.1367, -0.1377,  ...,  0.0806,  0.0206, -0.1064],
        ...,
        [ 0.0148,  0.0693,  0.1196,  ..., -0.0806, -0.1553, -0.1226],
        [-0.1299,  0.0630, -0.0128,  ...,  0.1514, -0.0515, -0.0272],
        [ 0.0928, -0.0381, -0.0476,  ..., -0.1475,  0.0315,  0.1377]],
       dtype=torch.bfloat16)
```

The `weight=not recognized` does not happen if quantization is not applied, but the weights don't seem to be AQT

Reply:
so for second repro it's expected, since we check for the type of the child modules, if you change 
```
model = torch.nn.Linear(1000, 1000)
```

to

```
model = torch.nn.Sequential(torch.nn.Linear(1000, 1000))
```
it should work I think.

for the first example, my guess is it's skipped because the shape did not match group_size: https://github.com/pytorch/ao/blob/e06fa8d250e8b21e1c9daca03f02030adb90d9a8/torchao/quantization/quant_api.py#L656-L660, because it's 32, 32, but group_size is 128. if you use group_size == 32, it should work.